src/stcal/saturation/saturation.py

```diff
@@ -85,6 +92,15 @@ def flag_saturated_pixels(data, gdq, pdq, sat_thresh, sat_dq, atod_limit,
             np.bitwise_or(gdq[ints, group, :, :], flaglowarray,
                           gdq[ints, group, :, :])
 
+            # now, flag any pixels that border saturated pixels (not A/D floor pix)
+            if n_pix_grow_sat > 0:
+                gdq_slice = copy.copy(gdq[ints, group, :, :]).astype(int)
+                only_sat = np.bitwise_and(gdq_slice, saturated).astype(np.uint8)
+                box_dim = (n_pix_grow_sat * 2) + 1
+                struct = np.ones((box_dim, box_dim)).astype(bool)
+                dialated = ndimage.binary_dilation(only_sat, structure=struct).astype(only_sat.dtype)
```

Comment:
I like this `binary_dilation` method (never seen it before). Is it possible to use it in a mode where you would send it the entire 4D groupdq array after the initial round of flagging is complete (i.e. the loops over groups and integrations is complete) and have it only expand the flags in the 3rd and 4th axes (i.e. image rows and cols)? That way it could be called just once, instead of within the loops here.